lib/icingadb/icingadb-objects.cpp

```diff
@@ -2226,6 +2228,20 @@ Dictionary::Ptr IcingaDB::SerializeState(const Checkable::Ptr& checkable)
 		}
 	}
 
+	{
+		Comment::Ptr lastComment;
+
+		for (auto& comment : checkable->GetComments()) {
```

Comment:
This will copy the `set<Comment::Ptr>` each time. Maybe this could be done within the `Checkable` class so that it has to only iterate the whole set but not allocate a second one.

Also, depending on the number of comments to expect on a single `Checkable`, maybe it would keep sense to keep track of them ordered by time so that this iteration isn't necessary each time. But probably not worth the effort if we don't regularly expect way more then maybe 100 comments.

Reply:
Do you consider s/set/vector/ a valid solution to reduce malloc() to a minimum?

Reply:
Why do you still want to perform a copy? `Comment::Ptr Checkable::GetLastComment()` that just does the very same loop holding the mutex should do the trick (will take about the same time as copying the set while holding the mutex, maybe even less).

Reply:
Do you consider the suggested solution valid or not?

Reply:
It still performs a useless copy that's discarded right away. So no, unless you can explain why that would be a better solution. Apart from that, you'd have to check all other callers and see if they rely on the return value being a set.